utils/build-apple-framework.sh

```diff
@@ -92,28 +100,18 @@ function build_apple_framework {
   fi
 }
 
-# Accepts an array of frameworks and will place all of
-# the architectures into the first one in the list
+# Merges compiled frameworks and combines all of
+# the architectures into the single xcframework
 function create_universal_framework {
   cd ./destroot/Library/Frameworks || exit 1
 
-  local platforms=("$@")
-
-  echo "Creating universal framework for platforms: ${platforms[*]}"
-
-  for i in "${!platforms[@]}"; do
-    platforms[$i]="${platforms[$i]}/hermes.framework/hermes"
-  done
```

Comment:
Anything in particular wrong with this code? Why not replace `"${platforms[$i]}/hermes.framework/hermes"` with `"${platforms[$i]}/hermes.framework"` and keep using it later?

That way, you could use `xcodebuild` in a similar fashion as I used `lipo`, `lipo -create -output "${platforms[0]}" "${platforms[@]}"`.

Reply:
I see the likely reason for replacing this and going with a hardcoded way was that we always build and include `macosx` platform now, with iOS's Catalyst project in mind.

However, there's one detail to keep in mind here that we may wanna take a closer look at. The caching. Right now, since both Catalyst and regular Mac builds share the same platform name "macosx", they will go into the same build folder. In other words, running iOS and Mac builds will produce different contents of "macosx" folder, potentially introducing conflicts and build issues when run together. It will also make it much harder to debug Hermes on both Mac (application) and iOS (Catalyst) at the same time, iOS build will invalidate Mac artifacts.

With that in mind, the cleanest solution would be to produce `build_macosx` and `build_catalyst` build folders as well as update `destroot` contents so that they don't collide each other. 

To do so, we would need to update `configure_apple_framework` (I have pasted a special comment on that few lines earlier) and CMake to handle that case.

Reply:
As far as updating CMake configuration files, you need to go here https://github.com/facebook/hermes/blob/74595df507b2dd38a677c8cb8c7615ab6ccc61d0/API/hermes/CMakeLists.txt#L133 and change `FRAMEWORK DESTINATION Library/Frameworks/${HERMES_APPLE_TARGET_PLATFORM}` to be different when Catalyst is done. 

Since the names don't really matter for `xcodebuild`, I would suggest we produce `macosx` and `macosx_catalyst` folders inside `Library/Frameworks`, to go in line with the `build_macosx` and `build_macosx_catalyst`.

Then, the platform argument in the script would be `macosx` and `macosx_catalyst`. You would only need to tweak it inside `configure_apple_framework` function to set `DHERMES_APPLE_TARGET_PLATFORM` and `DHERMES_APPLE_CATALYST` in both scenarios. 

Once that is done, both `build_apple_framework` and `create_universal_framework` functions would need no special treatment for Catalyst, since "macosx_catalyst" would be a valid folder and would contain valid files.

Reply:
Once this change is done, we gonna have `catalyst` and `macosx` frameworks inside `destroot`. To keep this function universal, the aforementioned generic logic of not assuming any particular platform would need to be reverted.